pyqtgraph/jupyter/GraphicsView.py

```diff
@@ -0,0 +1,127 @@
+    "pointer_up" : QtCore.QEvent.Type.MouseButtonRelease,
+    "pointer_move" : QtCore.QEvent.Type.MouseMove,
+}
+
+def get_buttons(evt_buttons):
+    NoButton = QtCore.Qt.MouseButton.NoButton
+    btns = NoButton
+    for x in evt_buttons:
+        btns |= MBLUT.get(x, NoButton)
+    return btns
+
+def get_modifiers(evt_modifiers):
+    NoModifier = QtCore.Qt.KeyboardModifier.NoModifier
+    mods = NoModifier
+    for x in evt_modifiers:
+        mods |= KMLUT.get(x, NoModifier)
+    return mods
+
+
+class GraphicsView(jupyter_rfb.RemoteFrameBuffer):
+    def __init__(self, **kwds):
+        super().__init__(**kwds)
+        self.gfxView = widgets.GraphicsView.GraphicsView()
+        self.logical_size = 640, 480
+        self.pixel_ratio = 1.0
+        # self.gfxView.resize(*self.logical_size)
+        # self.gfxView.show()
+        # self.gfxView.resizeEvent(None)
+
+    def get_frame(self):
+        w, h = self.logical_size
+        dpr = self.pixel_ratio
+        buf = np.empty((int(h * dpr), int(w * dpr), 4), dtype=np.uint8)
+        qimg = fn.ndarray_to_qimage(buf, QtGui.QImage.Format.Format_RGBX8888)
+        qimg.fill(QtCore.Qt.GlobalColor.transparent)
+        qimg.setDevicePixelRatio(dpr)
+        painter = QtGui.QPainter(qimg)
+        self.gfxView.render(painter, self.gfxView.viewRect(), self.gfxView.rect())
+        painter.end()
+        return buf
+    
+    def handle_event(self, event):
+        event_type = event["event_type"]
+
+        if event_type == "resize":
+            self.logical_size = int(event["width"]), int(event["height"])
+            self.pixel_ratio = event["pixel_ratio"]
+            self.gfxView.resize(*self.logical_size)
+            self.gfxView.resizeEvent(None)
+        elif event_type in ["pointer_down", "pointer_up", "pointer_move"]:
+            btn = MBLUT.get(event["button"], None)
+            if btn is not None:    # ignore unsupported buttons
+                pos = QtCore.QPointF(event["x"], event["y"])
+                btns = get_buttons(event["buttons"])
+                mods = get_modifiers(event["modifiers"])
+                typ = TYPLUT[event_type]
+                evt = QtGui.QMouseEvent(typ, pos, pos, btn, btns, mods)
+                QtCore.QCoreApplication.sendEvent(self.gfxView.viewport(), evt)
+                self.request_draw()
+        elif event_type == "wheel":
+            pos = QtCore.QPointF(event["x"], event["y"])
+            pixdel = QtCore.QPoint()
+            scale = -1.0    # map JavaScript wheel to Qt wheel
+            angdel = QtCore.QPoint(int(event["dx"] * scale), int(event["dy"] * scale))
+            btns = get_buttons([])
+            mods = get_modifiers(event["modifiers"])
+            phase = QtCore.Qt.ScrollPhase.NoScrollPhase
+            inverted = False
+            evt = QtGui.QWheelEvent(pos, pos, pixdel, angdel, btns, mods, phase, inverted)
+            QtCore.QCoreApplication.sendEvent(self.gfxView.viewport(), evt)
+            self.request_draw()
```

Comment:
Typically the event will be emitted into the application, and if the application reacts to it with changes that need a redraw, it will call `request_draw`. Then again, this should probably work too.

Reply:
The implementation done here has jupyter_rfb using pyqtgraph as a library; rather than pyqtgraph using jupyter_rfb as a backend. i.e. pyqtgraph has no idea that it is being used in jupyter_notebook, and would not be able to call ```request_draw()```

I did initially start off with a class inheriting from both pyqtgraph's GraphicsView and RemoteFrameBuffer; but got this error with PySide6: ```TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases```